gui/test/ip.spec.ts

```diff
@@ -0,0 +1,117 @@
+import { expect } from 'chai';
+import { it, describe } from 'mocha';
+import * as ip from '../src/renderer/lib/ip';
+
+const validIpv4Addresses = [
+  '127.0.0.1',
+  '10.255.255.255',
+  '192.168.1.1',
+  '192.168.0.10',
+  '192.168.1.254',
+  '192.168.254.254',
+  '10.0.0.1',
+  '10.90.90.90',
+  '1.1.1.1',
+  '193.138.218.74',
+];
+
+const validIpv6Addresses = [
+  '0:1:2:3:4:5:6:7',
+  '00:11:22:33:44:55:66:77',
+  '000:111:222:333:444:555:666:777',
+  '0000:1111:2222:3333:4444:5555:6666:7777',
+  'ffff::',
+  '::ff:2233',
+  'fee::ff:2233',
+];
+
+const invalidIpv4Addresses = [
```

Comment:
Would it be good to have in test outside of range, something like 260.260.260.260 ?

Reply:
Thanks for the input! `10.0.0.256`, 256 is outside the range :)

Reply:
Please feel free to suggest more test cases. I haven't done a lot of IP validation/parsing/range comparisons before so I could definitely have made mistakes here.